tools/lint-sh.js

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env node
+'use strict';
+
+const { spawn } = require('child_process');
+const { promises: fs, readdirSync } = require('fs');
+const { extname, join, relative, resolve } = require('path');
+const DIR = resolve(process.argv[2]);
+
+const SHELLCHECK_EXE = process.env.SHELLCHECK || '/usr/local/bin/shellcheck';
+const GIT_EXE = process.env.GIT || '/usr/bin/git';
+const SHELLCHECK_OPTIONS = ['--shell=sh', '--severity=info', '--enable=all'];
+if (process.argv[3] === '--fix') SHELLCHECK_OPTIONS.push('--format=diff');
+else if (process.env.GITHUB_ACTIONS) SHELLCHECK_OPTIONS.push('--format=json');
+
+const SPAWN_OPTIONS = {
+  cwd: DIR,
+  shell: false,
+  stdio: ['pipe', 'pipe', 'inherit'],
+};
+
+function* findScriptFilesRecursively(dirPath) {
+  const entries = readdirSync(dirPath, { withFileTypes: true });
+
+  for (const entry of entries) {
+    const path = join(dirPath, entry.name);
+
+    if (
+      entry.isDirectory() &&
+      entry.name !== 'build' &&
+      entry.name !== 'changelogs' &&
+      entry.name !== 'deps' &&
+      entry.name !== 'fixtures' &&
+      entry.name !== 'gyp' &&
+      entry.name !== 'node_modules' &&
+      entry.name !== 'out' &&
+      entry.name !== 'tmp'
+    ) {
+      yield* findScriptFilesRecursively(path);
+    } else if (entry.isFile() && extname(entry.name) === '.sh') {
+      yield path;
+    }
+  }
+}
+
+const expectedHashBang = Buffer.from('#!/bin/sh\n');
```

Comment:
Some of our scripts use bashisms so it would be incorrect to use `#!/bin/sh` on those.

Reply:
My plan is to remove those bashisms, it doesn't seem we are using a lot of them.